models/models_methods.go

```diff
@@ -450,6 +470,74 @@ func (config *BitriseDataModel) Validate() ([]string, error) {
 	}
 	// ---
 
+	// pipelines
+	for ID, pipeline := range config.Pipelines {
+		if ID == "" {
+			return warnings, fmt.Errorf("invalid pipeline ID (%s): empty", ID)
+		}
+
+		r := regexp.MustCompile(`[A-Za-z0-9-_.]+`)
+		if find := r.FindString(ID); find != ID {
+			warnings = append(warnings, fmt.Sprintf("invalid pipeline ID (%s): doesn't conform to: [A-Za-z0-9-_.]", ID))
```

Comment:
Just to double check, shouldn't invalid pipelineID be an error?

Reply:
Nope, just like for workflows, the ID validation works separately from the pipeline/stage/workflow model validation, not returning error but instead just a warning.